packages/playground/src/components/select_gpu_node.vue

```diff
@@ -0,0 +1,183 @@
+  name: string;
+  disk: number;
+  disks: {
+    name?: string | undefined;
+    size: number;
+    mountPoint: string;
+  }[];
+}
+
+const emits = defineEmits<{ (event: "update:modelValue", value?: GPUNodeType): void }>();
+
+const props = defineProps({
+  modelValue: { type: Object as PropType<GPUNodeType> },
+  filters: { default: () => ({} as GPUMachineFilters), type: Object as PropType<GPUMachineFilters> },
+});
+
+const profileManager = useProfileManager();
+const availableNodesWithGPU = ref<Array<number>>([]);
+const loadingNodes = ref(false);
+const loadingCards = ref(false);
+const selectedNode = ref<number>();
+const selectedCards = ref<Array<string>>([]);
+const nodeCards = ref<Array<NodeGPUCardType>>([]);
+
+watch(selectedCards, async () => {
+  const cards: NodeGPUCardType[] = [];
+  for (const card of nodeCards.value) {
+    for (const selectedCard of selectedCards.value) {
+      if (card.vendor === selectedCard && !cards.includes(card)) {
+        cards.push(card);
+      }
+    }
+  }
+  if (selectedNode.value && selectedCards.value) {
+    emits("update:modelValue", { nodeId: selectedNode.value, cards: cards });
+  }
+});
+
+watch(selectedNode, async () => {
+  loadingCards.value = true;
+  const projectName = ProjectName.Fullvm.toLowerCase();
+  const grid = await getGrid(profileManager.profile!, projectName);
+  if (grid) {
+    const gpuNodeHelper = new GPUNode(grid);
+    if (selectedNode.value) {
+      gpuNodeHelper.getNodeCards(selectedNode.value).then(res => {
+        nodeCards.value = res.filter(card => card.contract == 0);
+        loadingCards.value = false;
+      });
+    }
+  }
+});
+
+watch(
+  () => props.filters.ipv4,
+  async () => {
+    checkNode();
+  },
+);
+
+onMounted(checkNode);
+
+async function checkNode() {
+  const filters = props.filters;
+  if (filters.hasGPU) {
+    const projectName = ProjectName.Fullvm.toLowerCase();
+    const grid = await getGrid(profileManager.profile!, projectName);
+
+    if (grid) {
+      const gpuNodeHelper = new GPUNode(grid);
+      loadingNodes.value = true;
+      gpuNodeHelper
+        .getNodes({
+          name: filters.name,
+          machines: [
+            {
+              name: filters.name,
+              cpu: filters.cpu,
+              memory: filters.memory,
+              flist: filters.flist!.value,
+              entryPoint: filters.flist!.entryPoint,
+              disks: [{ size: filters.disk, mountPoint: "/" }, ...filters.disks],
+              publicIpv4: filters.ipv4,
+              publicIpv6: filters.ipv6,
+              planetary: filters.planetary,
+              envs: [{ key: "SSH_KEY", value: profileManager.profile!.ssh }],
+              rootFilesystemSize: 2,
+              hasGPU: filters.hasGPU,
+              rentedBy: filters.hasGPU ? profileManager.profile!.twinId : undefined,
+            },
+          ],
+          network: { addAccess: filters.wireguard },
+        })
+        .then(res =>
+          res.forEach(nodes => {
```

Comment:
Please catch the possible errors and handle them by showing them to the user under the vuetify input

Reply:
Done updated